script.js

```diff
@@ -1,4 +1,381 @@
+  }
+  return deck;
+};
+
+// Assign Card Names (ace, jack, queen,king)
+// insert rank into input
+var assignCardNames = function (input) {
+  if (input == 1) {
+    return "ace";
+  } else if (input == 11) {
+    return "jack";
+  } else if (input == 12) {
+    return "queen";
+  } else if (input == 13) {
+    return "king";
+  } else {
+    return String(input);
+  }
+};
+
+// Assign Card Numbers (1,11,12,13 -> Number 1 and 10)
+var assignCardNumbers = function (input) {
+  if (input == 1) {
+    return 1;
+  } else if (input == 11) {
+    return 10;
+  } else if (input == 12) {
+    return 10;
+  } else if (input == 13) {
+    return 10;
+  } else {
+    return input;
+  }
+};
+
+// Shuffle a deck - Inputs: Deck (array type)
+var shuffleDeck = function (input) {
+  var i = 0;
+  while (i < input.length) {
+    var randomNumber = randomNum();
+    var proxycard_i = deck[i];
+    input[i] = input[randomNumber];
+    input[randomNumber] = proxycard_i;
+    i += 1;
+  }
+  return input;
+};
+
+// Get a random number
+var randomNum = function () {
+  return Math.floor(Math.random() * 52);
+};
+
+// GAME BEGINS HERE
+
+var gamestate = "START_STATE"; // START_STATE, HIT_STATE, STAND_STATE
+var playerHand = [];
+var computerHand = [];
+var deck = makeDeck();
+var shuffledDeck = shuffleDeck(deck);
+var playerHandValue = 0;
+var computerHandValue = 0;
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  //check gamestate
+  if (checkCorrectInput(input)) {
+    return `Please enter Hit or Stand`;
+  } else {
+    checkGameState(input);
+  }
+  console.log("Check initial game state:" + gamestate);
+
+  if (gamestate == "START_STATE") {
+    gamestate = "PENDING_PLAYER_DECISION";
+    return startInitialGame();
+  }
+
+  if (gamestate == "HIT_STATE") {
+    var output = init_hit_state();
+    // console.log("Player hits, 1 card drawn, check deck: ");
+    // console.log(shuffledDeck);
+    return output;
+  } else if (gamestate == "STAND_STATE") {
+    console.log("Reach stand state");
+    init_stand_state();
+
+    var outcome = compareHandValues(playerHand, computerHand);
+    var myOutPut = GameOutcome(outcome);
+    resetgame();
+    return myOutPut;
+  }
+};
+
+/// 2. GAME STATE FUNCTION
+// Initial game state
+// Draw 2 cards for player and computer
+// Updates the drawn deck
+// Check for blackjack conditions
+var startInitialGame = function () {
+  for (var i = 0; i < 2; i += 1) {
+    var playercard = drawCard(shuffledDeck);
+    playerHand.push(playercard);
+    var computercard = drawCard(shuffledDeck);
+    computerHand.push(computercard);
+    console.log(playercard);
+    console.log(computercard);
+  }
+  //check for blackjack first
+  checkBlackJack(playerHand);
+  return hitOrStand();
+};
+
+// Hit state
+var init_hit_state = function () {
+  // to check if the deck is updated
+  var playercard = drawCard(shuffledDeck);
+  playerHand.push(playercard);
+  if (checkHandSum(playerHand) > 21) {
```

Comment:
since this is a blackjack game, it'll be common for us to check if a hand has gone above 21. So a helper function would be ideal here
```js
function isBusted(playerHand) {
// logic here
}
```